runtime/lua/vim/url.lua

```diff
@@ -0,0 +1,23 @@
+local M = {}
+
+--- Gets the URL at cursor, if any.
+function M._get_url()
```

Comment:
either `vim.ui` or `vim.text` seem like a better place for this. probably `vim.ui`, since that is where most provider-like features should live.